src/cryptsetup/cryptsetup-generator.c

```diff
@@ -106,11 +152,33 @@ static int create_disk(
                 "DefaultDependencies=no\n"
                 "Conflicts=umount.target\n"
                 "IgnoreOnIsolate=true\n"
-                "After=%s\n",
+                "After=%s\n\n",
                 netdev ? "remote-fs-pre.target" : "cryptsetup-pre.target");
 
+        if (keydev) {
+                _cleanup_free_ char *unit = NULL, *mount = NULL, *p = NULL;
+
+                if (!password)
+                        return log_error_errno(-EINVAL, "Path to password file not specified: %m");
```

Comment:
I'd prefer to have this check earlier, before the output file is opened, so we don't leave a half-written file on error.

Reply:
But, we leave a half-written file on other error conditions too. Maybe we should somehow refactor other places too. Possibly use memory stream instead of the file and flush the stream to file at the very end.

Reply:
But in this case it's very easy to dtrt. Let's just do this case, and worry about the rest some other time.

Reply:
Fair enough, I moved the check before we create the file. I also slightly modified the error message.